superset/columns/models.py

```diff
@@ -0,0 +1,95 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Column model.
+
+This model was introduced in SIP-68 (https://github.com/apache/superset/issues/14909),
+and represents a "column" in a table or dataset. In addition to a column, new models for
+tables, metrics, and datasets were also introduced.
+
+These models are not fully implemented, and shouldn't be used yet.
+"""
+
+import sqlalchemy as sa
+from flask_appbuilder import Model
+
+from superset.models.helpers import (
+    AuditMixinNullable,
+    ExtraJSONMixin,
+    ImportExportMixin,
+)
+
+
+class Column(
+    Model, AuditMixinNullable, ExtraJSONMixin, ImportExportMixin,
+):
+    """
+    A "column".
+
+    The definition of column here is overloaded: it can represent a physical column in a
+    database relation (table or view); a computed/derived column on a dataset; or an
+    aggregation expression representing a metric.
+    """
+
+    __tablename__ = "sl_columns"
+
+    id = sa.Column(sa.Integer, primary_key=True)
```

Comment:
not a blocker, but more of a question.. when do we want to start using uuids instead of auto-incrementing integers for  public-facing ids?

Reply:
I think we would need a SIP and a major release for that, no?

Reply:
Since this is a new model, I don't see any reason why we couldn't add it in here since it wouldn't break anything, but yes I agree on a SIP regarding an overall change toward that methodology. If that's the blocker, then that makes sense.

Reply:
Ah, I see your point. But the idea is that at least initially the APIs would be unchanged, we'd just reroute the backend to read from the new models instead of the old ones.